src/WebJobs.Script/ManagedDependencies/ManagedDependencyProvider.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.PowerShellWorker.ManagedDependency;
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.ManagedDependencies
+{
+    internal class ManagedDependencyProvider : IManagedDependencyProvider
+    {
+        private const string ManagedDependenciesPropertiesName = "managedDependencies";
+
+        /// <summary>
+        /// Returns the managed dependecies meta data
+        /// </summary>
+        /// <param name="hostFilePath">The configuration path for the managed dependencies</param>
+        /// <param name="oSPlatform">The oSPlatform the code is running on</param>
+        /// <param name="managedDependencyConfigs">Configurations for managed dependencies</param>
+        /// <returns>The IEnumerable list of <see cref="Dependency"/></returns>
+        public async Task<IEnumerable<Dependency>> GetDependencies(string hostFilePath,
+            OSPlatform oSPlatform,
+            IManagedDependencyConfigs managedDependencyConfigs)
+        {
+            if (string.IsNullOrWhiteSpace(hostFilePath)
+                || !File.Exists(hostFilePath))
+            {
+                return null;
+            }
+
+            var managedDependenciesFromHostFile = await DesirializeJsonFile<DependentLibraries>(hostFilePath);
+            if (managedDependenciesFromHostFile.ManagedDependencies == null
+                || !managedDependenciesFromHostFile.ManagedDependencies.Any())
+            {
+                return null;
+            }
+
+            var dependencyInstallationList = new List<Dependency>();
+            foreach (var dependentLibrary in managedDependenciesFromHostFile.ManagedDependencies)
+            {
+                var dependency = new Dependency();
+                var moduleName = dependentLibrary.Name;
+                var majorVersion = dependentLibrary.MajorVersion;
+
+                dependency.Name = moduleName;
+                dependency.Majorversion = majorVersion;
+
+                var isDownloadRequired = false;
+                var dependencyDownloadPath = default(string);
+
+                //If platform is windows and app service environment
+                if (oSPlatform == OSPlatform.Windows
+                    && !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId)))
```

Comment:
Avoid direct use of `Environment` and take an `IEnvironment` instead. We have extension methods on that interface that will perform all of those checks for you as well.

Reply:
You can use this method instead `IsAppServiceWindowsEnvironment` from https://github.com/Azure/azure-functions-host/blob/12d3f4b976b03035c2ce93030f414d9a613bc523/src/WebJobs.Script/Environment/EnvironmentExtensions.cs#L65